cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/default-service/AwsDefaults.service.tsx

```diff
@@ -25,4 +32,39 @@ export class AwsDefaults {
         }
         return nodeTypeList[Math.round(nodeTypeList.length / 2)];
     };
+
+    static defaulAvailabilityZoneNameStrategy = (azList: { [key: string]: string }[], nodeProfile: string) => {
+        const defaultAZNameList: string[] = [];
+        switch (nodeProfile) {
+            case AWS_NODE_PROFILE_NAMES.SINGLE_NODE: {
+                return [azList[0].id];
+            }
+            default: {
+                azList.slice(0, 3).forEach((az) => {
```

Comment:
Even though we're "guaranteed" that the azList should have at least 3 elements, I would still test it here and log an error if it's too small (rather than blow up)

Reply:
Yes sir. Add a check.